api/src/routing.js

```diff
@@ -162,6 +163,7 @@ app.use(
     ':res[content-length] :response-time ms'
   )
 );
+app.use(rateLimiterMiddleware);
```

Comment:
Right. Because you can basic auth to any endpoint we have to watch everything. But only 401s and 429s end up consuming points.

The downside is a small tax on every request. At the moment it's an in memory lookup by key so it's supposed to be really fast.

An alternative would be to only do the lookup if someone is logging in either with basic auth or with the login api but quickly skip for every other request. This would have a slight performance benefit, but also allow things like static resources to be returned even when you've hit your rate limit.

What do you think?

Reply:
I'm not worried about the small tax, I'm worried about requests that should be honored (like static resources, great example) even when maybe rate limited. So I'm leaning to simply skipping the check for requests that don't carry any login information.

Reply:
Yeah I'm maybe leaning that way too :thinking: 

For a laugh you can look at all the early commits on this branch. I went back and forwards so many times about the correct way to do it without breaking legitimate users, and this may be a hangover from a previous more aggressive approach.

I think about it some more and get back to you...

Reply:
Done.